packages/protocol/script/DeployERC20Airdrop.s.sol

```diff
@@ -16,39 +17,120 @@ import "../contracts/team/airdrop/ERC20Airdrop.sol";
+
+    /// @notice Address of the bridged TKO token
+    /// @dev The address must be valid and the TKO token must already exist for the deployment to
+    /// proceed
     address public bridgedTko = vm.envAddress("BRIDGED_TKO_ADDRESS");
-    address public vaultAddress = vm.envAddress("VAULT_ADDRESS");
 
+    uint256 public airdropStartTime = vm.envUint("AIRDROP_START_TIME");
+
+    uint256 public airdropEndTime = vm.envUint("AIRDROP_END_TIME");
+
+    /// @notice Private key of the deployer used for transactions
+    /// @dev This private key must be set in the environment variables and is critical for contract
+    /// deployment
+    bytes32 public merkleRoot = vm.envBytes32("AIRDROP_MERKLE_ROOT");
+    ERC20Airdrop airdropContract;
+
+    /// @notice Prepares the contract setup for deployment
+    /// @dev This function checks for the presence of the bridgedTko and vaultAddress.
+    /// If absent in a testnet environment, it deploys new instances.
     function setUp() external { }
 
+    function testEnvOnly() internal {
+        // Testnet Environment
+        if (block.chainid != 167_000) {
+            console.log("=== Testnet Environment ===");
+            // Deploy a mock ERC20 token
+            console.log("[TEST] Deploying mock TKO token");
+            MockERC20 mockERC20 = new MockERC20();
+            // Replace bridgedTko with the mock ERC20 token
+            console.log("[TEST] TKO token set as", address(mockERC20));
+            bridgedTko = address(mockERC20);
+        }
+    }
+
+    function postDeployment() internal {
+        console.log("=== Post deployment ===");
+
+        if (block.chainid != 167_000) {
+            // If testnet, mint 50 Million tokens to the vault
+            console.log("[TEST] Minting 50 Million tokens to the airdrop contract");
+            MockERC20(bridgedTko).mint(address(airdropContract), 50_000_000_000e18);
+            console.log("[INFO] Setting Airdrop Config");
+            console.log(" - Airdrop Start Time:", airdropStartTime);
+            console.log(" - Airdrop End Time:", airdropEndTime);
+            console.log(" - Merkle Root:");
+            console.logBytes32(merkleRoot);
+
+            // Set Merkle Proof, Start Time and End Time
+            airdropContract.setConfig(uint64(airdropStartTime), uint64(airdropEndTime), merkleRoot);
+        } else {
+            // Transfer ownership to 0xf8ff2AF0DC1D5BA4811f22aCb02936A1529fd2Be
+            // This address will have ownership of the airdrop contract which will have the ability
+            // to set the airdrop start and end time, and the merkle root with
+            // setConfig(_claimStart, _claimEnd, _merkleRoot).
+            console.log(
+                "[INFO] Transferring Ownership of Airdrop Contract to 0xf8ff2AF0DC1D5BA4811f22aCb02936A1529fd2Be"
+            );
+            airdropContract.transferOwnership(0xf8ff2AF0DC1D5BA4811f22aCb02936A1529fd2Be);
```

Comment:
Maybe not hardcode but this address shall come from env var, or smth.. ? Cannot recall what 0xf8 is anymore :D